### PR TITLE
Pwgpp 576 bootstrap

### PR DESCRIPTION
**Pull request by pl0xz0rz**
This PR:

Adds benchmark_bootstrap.py, which runs curve_fit_BS with pytorch/tensorflow/scipy

Modifies @MartinKroe FitterBFGS, making two changes to the interface:
1. The default loss function accepts weights, it's only a valid MSE if the mean of weights is 1
2. curve_fit_BS now outputs a Pandas dataframe

Modifies data.py to generate float32 data instead of float64 data (@MartinKroe 's fitter was already casting data to float32)

Modifies benchmarkScanPlot.ipynb to work with the output of benchmark_bootstrap.py - will probably revert this change and instead add another jupyter notebook

### CODE REVIEW

**Comment by miranov25** [comment]
Hello Marian and Martin
I have several comments, what should be changed:

- I tried to run the code but it is not responding.  I prefer to print something after each fit
- I checked also Jupyter notebook - benchmarkScanPlot.ipynb 
   - We should provide full report about quality
  -  As we described in JIRA  https://alice.its.cern.ch/jira/browse/PWGPP-576?focusedCommentId=248471&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-248471

We can meet tonight on Vidyo, to finish benchmark. 
I suggest to meet oday at 20:00 in my Vidyo room. In case of interest Martin can also join.

**Comment by miranov25** [comment]
comment to pull request  -https://github.com/miranov25/RootInteractiveTest/pull/30
* time should be in table 1
* error to be added to table 1
* printout  to be  added in benchmark  as described in:
 https://alice.its.cern.ch/jira/browse/PWGPP-576?focusedCommentId=248885&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-248885

## print values, error of values and expected values+ Alarms

Alarms should be parameterized as n sigma deviation from expectation 

### Normal fit with init weigths  - All params,  and errors  + true value + corresponding plots 

- Average of  N  <fits-true>  with weight one  - expected value is 0    
- error of <fits-true> ~ rmsexp/sqrt(N)
- RMS of N <fits-true>  with with weight one  - expected value covariance estimate  
- Pulls <fit-true>/rmsexp
### Bootstrap:
* Timing estimate - sum of wall time and other estimators
* Mean/Median/RMS resp.  user defined  bootstrap  
* Bootstrap error - and covariance matrix
* RMS of distribution should be the same as from covariance matrix for gaus + chi2 cost function

**Comment by miranov25** [comment]
# Current status and open problems:
* bootstrap does not work properly yet in TensorFlow - spread is too small 
  * looks like always the same sample used
* Pytorch error estimate is overestimated~ 30 times 
  * Working hypothesis - looks like sqrt(Nsample)

**Comment by miranov25** [comment]
Merging version to master. Problems above to be solved later
